pkg/workloads/spark_job/spark_util.py

```diff
@@ -223,8 +223,30 @@ def ingest(ctx, spark):
 
     for raw_column_name in ctx.raw_columns.keys():
         raw_column = ctx.raw_columns[raw_column_name]
-        expected_types = CORTEX_TYPE_TO_ACCEPTABLE_SPARK_TYPES[raw_column["type"]]
         actual_type = input_type_map[raw_column_name]
+
+        if actual_type not in SPARK_TYPE_TO_CORTEX_TYPE.keys():
+            actual_type = StringType()
+
+        column_type = raw_column["type"]
+        if column_type == consts.COLUMN_TYPE_INFERRED:
+            TEST_DF_SIZE = 10
+            sample_df = df.select(raw_column_name).limit(TEST_DF_SIZE).collect()
```

Comment:
This is being executed for each raw column, but should only have to be executed once